### PR TITLE
[a11y] Pass the id of the node whose layout changed to accessibility controllers

### PR DESCRIPTION
**Pull request by m-sasha**
Currently, when the layout of a single node changes, we call onSemanticsChange on (desktop) `AccessibilityController` and (iOS) `AccessibilityMediator`, which causes them to re-sync the entire node tree.

## Proposed Changes

Pass the semantics id of the node whose layout changed to the accessibility controllers, so they can invalidate the tree more granularly. Note that this PR only passes the information; the actual granular handling will be done in separate PR(s).

## Testing

Test: Manually, only on the desktop.

### CODE REVIEW

File: compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/PlatformContext.skiko.kt
Comment (MatkovIvan) [inline compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/PlatformContext.skiko.kt:137]: We already discussed this with @elijah-semyonov and as far as I remember we decided to introduce optional second parameter to `onSemanticsChange` instead of adding a new callback

**Review by MatkovIvan — CHANGES_REQUESTED** [review]
Blocking to agree on API change before merging

File: compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/PlatformContext.skiko.kt
Comment (m-sasha) [reply compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/PlatformContext.skiko.kt:137]: Having it in a separate function adds more information; namely that only the bounds of the node have changed, and not any semantic properties.

Comment (m-sasha) [reply compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/PlatformContext.skiko.kt:137]: The API changed here isn't public, by the way, so no need for the "changes in API" tag.

Comment (MatkovIvan) [reply compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/PlatformContext.skiko.kt:137]: it's public but with "internal" annotation. The rule for tag is basically - "changes in `desktop/ui.api` file".
I'd apply the rule regarding two reviewers here

Comment (m-sasha) [reply compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/PlatformContext.skiko.kt:137]: I disagree. @InternalApi implies that it's internal, and only technically public because of the limitations of the language. We don't make any compatibility guarantees about such code.

Comment (MatkovIvan) [reply compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/PlatformContext.skiko.kt:137]: ok, it's not a big deal in this case

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/scene/ComposeSceneMediator.uikit.kt
Comment (MatkovIvan) [inline compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/scene/ComposeSceneMediator.uikit.kt:165]:
```suggestion
        val (_, accessibilityMediator) = current?.takeIf { it.first == semanticsOwner } ?: return
        accessibilityMediator.onLayoutChange(nodeId = semanticsNodeId)
```

**Review by MatkovIvan — APPROVED** [review]

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/scene/ComposeSceneMediator.uikit.kt
Comment (m-sasha) [reply compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/scene/ComposeSceneMediator.uikit.kt:165]: I agree, but I'll leave it to @elijah-semyonov to make this change in a separate PR, if he wants to. I was just copying the style from the function above.

**Review by elijah-semyonov — APPROVED** [review]

**Pull request merged**
